pkg/sql/pgwire/pgwire_test.go

```diff
@@ -2339,3 +2340,28 @@ func TestFailPrepareFailsTxn(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestOverlyLargeRequest(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	s, _, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer s.Stopper().Stop(context.TODO())
+
+	pgURL, cleanupFn := sqlutils.PGUrl(t, s.ServingAddr(), t.Name(), url.User(security.RootUser))
+	defer cleanupFn()
+
+	db, err := gosql.Open("postgres", pgURL.String())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	stmt, err := db.Prepare("select $1::text")
+	assert.NoError(t, err)
+
+	// This string will cause the message to exceed pgwire.maxMessageSize.
+	largeString := make([]byte, 128*1024*1024)
```

Comment:
I'd also like to see something just below the 128MB size succeed.

Reply:
Done.